src/index.rs

```diff
@@ -20,11 +21,68 @@ pub struct ResourceIndex {
 
 #[derive(Debug)]
 pub struct IndexUpdate {
-    pub deleted: HashSet<ResourceId>,
+    pub updated: HashMap<CanonicalPathBuf, ResourceMeta>,
```

Comment:
An update is deletion followed by addition, because `ResourceId` changes, so technically new version is a new resource. Two fields `added` and `deleted` should be enough. This is different in ARK Navigator which should be considered older version.